app/models/secret_storage.rb

```diff
@@ -0,0 +1,79 @@
+require 'attr_encrypted'
+
+module SecretStorage
+  class DbBackend
+    class Secret < ActiveRecord::Base
+      self.table_name = :secrets
+      self.primary_key = :id # uses a string id
+
+      ENCRYPTION_KEY = Rails.application.secrets.secret_key_base
+
+      attr_encrypted :value, key: ENCRYPTION_KEY, algorithm: 'aes-256-cbc'
+
+      before_validation :store_encryption_key_sha
+      validates :id, :encrypted_value, :encryption_key_sha, presence: true
+      validates :id, format: /\A\S+\/\S*\Z/
+
+      private
+
+      def store_encryption_key_sha
+        self.encryption_key_sha = Digest::SHA2.hexdigest(ENCRYPTION_KEY)
+      end
+    end
+
+    def self.read(key)
```

Comment:
Return an `OpenStruct` rather than a Hash, so you can say `secret.value` on the response?